### DISCUSSION TITLE
Cancel operation timed out

### DISCUSSION THREAD

**Discussion by moon-rui** [dbeaver/dbeaver#19398] [General]
### Description

Press the cancel button after execute a long SQL.

after about 5 seconds the error log show: org.jkiss.dbeaver.model.exec.DBCException: Cancel operation timed out

However, in the background, the SQL process did be killed.

### DBeaver Version

Community Edition 22.3.5

### Operating System

Windows 7

### Database and driver

Clickhouse jdbc 0.2.6

### Additional context

  \\

**Comment by VitaliyHula** [comment]
The same bug in MacOS: Ventura 13.1; DBeaver Version 23.0.0.202303040621; Database: PG

**Comment by ShadelessFox** [comment]
This is expected behavior.

When DBeaver cannot cancel the query by executing driver API, DBeaver kills it after a given timeout, throwing an expectation to signal what happened.

**Comment by Nimit-Shah** [comment]
Any update on this? How can i disable dbeaver SQL engine to not close the jdbc connection if i cancel the query? It seems that connection close takes forever and times out. 
Database: redshift.
